source/backends/debian/debpkgindex.d

```diff
@@ -63,65 +63,114 @@ public:
+    private immutable(string[]) findTranslations (const string suite, const string section)
     {
-        immutable enDescPath = buildPath ("dists", suite, section, "i18n", "Translation-en.%s");
-        string enDescFname;
+        import std.regex : ctRegex, matchFirst;
+
+        immutable inRelease = buildPath (rootDir, "dists", suite, "InRelease");
+        auto regex = ctRegex!(r"Translation-(\w+)$");
+        auto ret = appender!(string[]);
 
         try {
-            enDescFname = downloadIfNecessary (rootDir, tmpDir, enDescPath);
-        } catch (Exception e) {
-            logDebug ("No long descriptions for %s/%s: %s", suite, section, e.msg);
-            return;
+            const inReleaseContents = getFile (inRelease);
+
+            foreach (const ref entry; inReleaseContents) {
+                auto match = entry.matchFirst(regex);
+
+                if (match.empty)
+                    continue;
+
+                if (!ret.data.canFind (match[1]))
+                    ret.put (match[1]);
+            }
+        } catch (Exception ex) {
+            logWarning ("Couldn't download %s, will assume 'en' is available.", inRelease);
+            return ["en"];
         }
 
-        auto tagf = new TagFile ();
-        tagf.open (enDescFname);
+        return cast (immutable) ret.data;
+    }
 
-        logDebug ("Opened: %s", enDescFname);
-        do {
-            auto pkgname = tagf.readField ("Package");
-            auto rawDesc  = tagf.readField ("Description-en");
-            if (!pkgname)
-                continue;
-            if (!rawDesc)
-                continue;
+    private void loadPackageLongDescs (DebPackage[string] pkgs, string suite, string section)
+    {
+        immutable langs = findTranslations (suite, section);
+
+        foreach (const ref lang; langs) {
+            string fname;
+
+            immutable fullPath = buildPath ("dists",
+                                            suite,
+                                            section,
+                                            "i18n",
+                                            /* here we explicitly substitute a
+                                             * "%s", because
+                                             * downloadIfNecessary will put the
+                                             * file extension there */
+                                            "Translation-%s.%s".format(lang, "%s"));
+
+            try {
+                fname = downloadIfNecessary (rootDir, tmpDir, fullPath);
+            } catch (Exception ex) {
+                logDebug ("No long descriptions for %s/%s", suite, section);
+                return;
+            }
 
-            auto pkgP = (pkgname in pkgs);
-            if (pkgP is null)
-                continue;
+            auto tagf = new TagFile ();
+            try {
+                tagf.open (fname);
+            } catch (Exception e) {
```

Comment:
That catch->throw block can be omitted, if an exception happens here it will be forwarded anyway, so no need to be explicit about it.